app/models/output_state.rb

```diff
@@ -16,11 +16,28 @@ class OutputState < ApplicationRecord
   # +----------------------+
   validates :output, presence: true, if: :mass_validation_enabled?
 
-  validates :state, inclusion: { in: [
+  # Current values of state are:
+  # +----------------+
+  # | state          |
+  # +----------------+
+  # | CHECKED        |
+  # | FAILED         |
+  # | LOADED         |
+  # | LOADING        |
+  # | LOADING_ERROR  |
+  # | LOADING_QUEUED |
+  # | NULL           |
+  # | RUNNING        |
+  # | UNKNOWN        |
+  # +----------------+
+  validates :state, allow_nil: true, inclusion: { in: [
+    PipelineRun::STATUS_CHECKED,
+    PipelineRun::STATUS_FAILED,
     PipelineRun::STATUS_LOADED,
-    PipelineRun::STATUS_UNKNOWN,
     PipelineRun::STATUS_LOADING,
-    PipelineRun::STATUS_LOADING_QUEUED,
     PipelineRun::STATUS_LOADING_ERROR,
+    PipelineRun::STATUS_LOADING_QUEUED,
+    PipelineRun::STATUS_RUNNING,
+    PipelineRun::STATUS_UNKNOWN,
```

Comment:
Hey Greg, I couldn't find any evidence that STATUS_RUNNING and STATUS_CHECKED are valid output states.

Comment from pipeline_run.rb:
The output_states indicate the state of each target output, the progression being as follows:
UNKNOWN -> LOADING_QUEUED -> LOADING -> LOADED / FAILED 

(the docs are missing LOADING_ERROR)

Reply:
I think the only one that needs to be added is STATUS_FAILED.

Reply:
> I couldn't find any evidence that STATUS_RUNNING and STATUS_CHECKED are valid output states

The states I guess are legacy. They are only found older than 2019. Also `NULL`. 

Good investigation!